### PR TITLE
RC-14 and fixing failing test

### PR DESCRIPTION
**Pull request by LGLO**
Test for laziness were failing.

### CODE REVIEW

File: src/test/scala/zio/interop/JavaSpec.scala [hunk 1]
```diff
@@ -34,18 +34,18 @@ object JavaSpec {
       },
       testM("return an `IO` that produces the value from `Future`") {
         val someValue: UIO[Future[Int]] = UIO.effectTotal(CompletableFuture.completedFuture(42))
-        assertM(ZIO.fromFutureJava(someValue).run, succeeds(equalTo(42)))
+        assertM(ZIO.fromFutureJava(someValue), equalTo(42))
```
Comment (LGLO) [inline src/test/scala/zio/interop/JavaSpec.scala:37]: This is unsolicited but I just remove `.run` and `succeeds` - for success case I think it's natural to assert on value.

File: src/test/scala/zio/interop/JavaSpec.scala [hunk 2]
```diff
@@ -184,6 +184,7 @@ object JavaSpec {
           fiberClient  <- taskClient.fork
           resultServer <- fiberServer.join
           resultClient <- fiberClient.join
+          _            <- ZIO.effectTotal(server.close())
```
Comment (LGLO) [inline src/test/scala/zio/interop/JavaSpec.scala:187]: Without this subsequent run of `test` inside sbt session causes tests to error w/o message.

File: src/test/scala/zio/interop/JavaSpec.scala [hunk 3]
```diff
@@ -15,7 +15,7 @@ object JavaSpec {
       testM("be lazy on the `Future` parameter") {
         var evaluated         = false
         def ftr: Future[Unit] = CompletableFuture.supplyAsync(() => evaluated = true)
-        assertM(ZIO.fromFutureJava(UIO.effectTotal(ftr)).as(evaluated).run, succeeds(isFalse))
+        assertM(ZIO.fromFutureJava(UIO.effectTotal(ftr)).when(false).as(evaluated), isFalse)
```
Comment (LGLO) [inline src/test/scala/zio/interop/JavaSpec.scala:18]: In RC14 `as` become lazy on it's param, IMO uncovering race condition in test.
`ftr` was run, but `as(evaluated)` is computed as `false` during construction of effect and `ftr` is run when executing effect.

File: src/test/scala/zio/interop/JavaSpec.scala (hunk 1 above)
Comment (mijicd) [reply src/test/scala/zio/interop/JavaSpec.scala:37]: Makes sense.

**Review by mijicd — APPROVED** [review]

**Pull request merged**